kmcd-props.md

```diff
@@ -14,17 +14,174 @@ module KMCD-PROPS
+```
+
+Ink of an ilk = Sum of all urn ink across all users for that ilk.
+
+Total debt = Sum of all debt across all users.
+
+```k
+    syntax Int ::= sumOfAllDebt(Map, Int) [function, functional]
+ // ------------------------------------------------------------
+    rule sumOfAllDebt( ADDR |-> DAI USERDAI, SUM)
+      => sumOfAllDebt( USERDAI, SUM +Int DAI)
+
+    rule sumOfAllDebt( _ |-> _ USERDAI, SUM ) => sumOfAllDebt( USERDAI, SUM ) [owise]
+
+    rule sumOfAllDebt(.Map, SUM) => SUM
+```
+
+Total vice = Sum of all sin across all users.
+
+```k
+    syntax Int ::= sumOfAllSin(Map, Int) [function, functional]
+ // ------------------------------------------------------------
+    rule sumOfAllSin( ADDR |-> SIN USERSIN, SUM)
+      => sumOfAllSin( USERSIN, SUM +Int SIN)
+
+    rule sumOfAllSin( _ |-> _ USERSIN, SUM ) => sumOfAllSin( USERSIN, SUM ) [owise]
+
+    rule sumOfAllSin(.Map, SUM) => SUM
+```
+
+Total dai of all users = CDP debt for all users and gem + system debt (vice)
+
+```k
+    syntax Rat ::= sumOfAllUserDebt( ilks: Map, urns: Map, sum: Rat) [function, functional]
+ // ---------------------------------------------------------------------------------------
+    rule sumOfAllUserDebt(
+             ILKID |-> ILK ILKS => ILKS,
+             URNS,
+             SUM => SUM +Rat (rate(ILK) *Rat sumOfUrnArt(URNS, ILKID, 0)) )
+
+    rule sumOfAllUserDebt(_ |-> _ ILKS => ILKS, URNS, SUM) [owise]
+
+    rule sumOfAllUserDebt(.Map, _, SUM) => SUM
+```
+
+Violations
+----------
+
+A violation occurs if any of the properties above holds.
+
+```k
+    syntax Bool ::= violated(List) [function, functional]
+ // -----------------------------------------------------
+    rule violated(EVENTS) => zeroTimePotInterest(EVENTS)
+                      orBool unAuthFlipKick(EVENTS)
+                      orBool unAuthFlapKick(EVENTS)
+                      orBool potEndInterest(EVENTS)
+```
+
+A violation can be checked using the Admin step `assert`. If a violation is detected,
+it is recorded in the state and execution is immediately terminated.
+
+```k
+    syntax AdminStep ::= "assert"
+ // -----------------------------
+    rule <k> (assert => .) ... </k>
+         <events> EVENTS </events>
+      requires notBool violated(EVENTS)
+
+    rule <k> assert ~> _ => . </k>
+         <events> EVENTS </events>
+         <violation> false => true </violation>
+      requires violated(EVENTS)
+```
+
+### Bounded Debt Growth
+
+The Debt growth should be bounded in principle by the interest rates available in the system.
+
+```k
+    syntax Bool ::= totalDebtBounded    ( List             ) [function]
+                  | totalDebtBoundedAux ( List , Rat , Rat ) [function]
+ // -------------------------------------------------------------------
+    rule totalDebtBounded(.List)                           => true
+    rule totalDebtBounded(ListItem(Measure(DEBT, _)) REST) => totalDebtBoundedAux(REST, DEBT, 1) // initial DSR 1
```

Comment:
Just read the DSR here instead.

Reply:
Actually we can't do this because the `<pot-dsr>` cell is not accurate at the time we're processing this entry.